design-proposals/initial-networking.md

```diff
@@ -0,0 +1,64 @@
+# Initial networking for VMs using pod NICs
+
+Author: Fabian Deutsch \<fabiand@redhat.com\>
+
+## Introduction
+
+VMs need network connectivity. This proposal is about a first start in this direction.
+
+The working assumption is that libvirtd is running with a host-view, thus in the host's
+network namespace.
+On the other hand we want to connect VMs to the NIC(s) of the pod in which they are running.
+
+To keep it simple and work with what we have today, we just handle a single NIC (thus multiple should
+not make a difference).
+
+Another benefit of this proposal is that we will connect to the veth of the pod - thus all the networking infrastructure beyong this point is not relevant to us. Which means we are independnt of it for now. It could be delivered by weave, flannel, or something else.
```

Comment:
Note that his implementation would be would be limited to solutions which use veth pairs. It is also possible to attach a macvlan to a host device and then directly move it into another namespace without veths between.